### PR TITLE
Remove ipex & pytorch recompilation if cached

### PR DESCRIPTION
**Pull request by leshikus**
`pip` from conda uses conda env to install packages, thus pip cache is not needed IMHO

### CODE REVIEW

**Comment by leshikus** [comment]
Temporarily closing to skip extra CI checks

File: .github/workflows/conda-build-test.yml [hunk 1]
```diff
@@ -105,32 +95,33 @@ jobs:
...
-          dest: ${{ steps.conda-cache.outputs.dest }}
+          conda run -n triton scripts/test-triton.sh
 
       - name: Run E2E test
         run: |
-          conda activate triton
+          cd ../pytorch || {
+            PYTORCH_COMMIT_ID=$(<.github/pins/pytorch.txt)
+            cd ..
+            git clone --single-branch -b dev/triton-test-3.0 --recurse-submodules https://github.com/Stonepia/pytorch.git
```
Comment (pbchekin) [inline .github/workflows/conda-build-test.yml:105]: What about using `actions/checkout` for this?

Comment (leshikus) [reply .github/workflows/conda-build-test.yml:105]: Will try using `actions/checkout`

Comment (leshikus) [reply .github/workflows/conda-build-test.yml:105]: I've got `Error: Repository path '/runner/_work/intel-xpu-backend-for-triton/pytorch' is not under '/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton'`

Thus I'll commit an original variant. We may move `pytorch` directory under `intel-xpu-backend-for-triton` or even make a recursive submodule to make `git` manage pins

**Pull request merged**